args.c

```diff
@@ -156,7 +162,7 @@ void parse_args(int argc, char* argv[])
 			argv_dnsmasq = calloc(argc_dnsmasq + 2,sizeof(char*));
 			argv_dnsmasq[0] = "";
 			if(debug) argv_dnsmasq[1] = "-d";
-			else      argv_dnsmasq[1] = "";
+			else      argv_dnsmasq[1] = "-k";
```

Comment:
Just a note: We set `-d / -k` here again a we re-allocated `argv_dnsmasq `

Reply:
Just for noting and documentations, this sets to not background when invoked in daemon mode. `--no-daemon` (-d) `--keep-in-foreground` (-k)

Reply:
Daemon (`-d`) implies not forking (`-k`) so replacing `-k` by `-d` is fine.